index.js

```diff
@@ -87,4 +87,75 @@ form.addEventListener('submit', (event) => {
   if (nameValid && emailValid) {
     form.submit();
   }
-});
\ No newline at end of file
+});
+
+// Store and preserve local data
+
+// Check if localStorage is supported by the browser
+function storageAvailable(type) {
+  let storage;
+  try {
+    storage = window[type];
+    const x = '__storage_test__';
+    storage.setItem(x, x);
+    storage.removeItem(x);
+    return true;
+  } catch (e) {
+    return (
+      e instanceof DOMException
+      // everything except Firefox
+      && (e.code === 22
+        // Firefox
+        || e.code === 1014
+        // test name field too, because code might not be present
+        // everything except Firefox
+        || e.name === 'QuotaExceededError'
+        // Firefox
+        || e.name === 'NS_ERROR_DOM_QUOTA_REACHED')
+      // acknowledge QuotaExceededError only if there's something already stored
+      && storage
+      && storage.length !== 0
+    );
+  }
+}
+
+function saveFormData() {
+  // Create an object to store the form data
+  const formData = {
+    name: document.getElementById('name').value,
+    email: document.getElementById('email').value,
+    message: document.getElementById('message').value,
+  };
+
+  // Convert formData to a JSON string
+  const jsonData = JSON.stringify(formData);
+
+  // Store JSON string in localStorage
+  localStorage.setItem('formData', jsonData);
```

Comment:
- [ ] Please 🙏🏻 adhere to the project requirement, according to which your data should be persistent if a user tends to refuse to submit the query and refresh the page or open the website after a while then it will form a field automatically pre-filled with the data where actually user left. But in your case, the data is saved to local storage but only when the user submits the query. The whole idea 💡 is here to save the user's time and effort where the form has multiple fields and having this kind of functionality where user data is persisted in localStorage on the client side actually really saves the user's time and maintains a good User Experience ✅. Any questions feel free to ask and for the hint, I have attached a video below.

## Requirement
![Screenshot from 2023-06-22 23-51-09](https://github.com/George7h/George7h.github.io/assets/66476329/cb36fa35-b23e-41bd-9ca7-a9319065d951)


## Video

[Screencast from 23-06-23 12:00:49 AM IST.webm](https://github.com/George7h/George7h.github.io/assets/66476329/f789195b-4a27-4719-97f2-77086af22284)